Readme.md: Adding Contributors guide (#5184)
juspay/hyperswitch: docs/CONTRIBUTING.md

[coc]: https://www.rust-lang.org/policies/code-of-conduct

Within the repositories, you'll find the following directories and files,
logically grouping common assets and providing both compiled and minified

> variations.

Reviewer: Why is this under the code of conduct? I'm guessing this was supposed to be under the `Repositories` section?

### Repositories

> The current setup contains a three different repos, which contains different Hyperswitch components.

Reviewer: ```suggestion
The current setup contains three different repositories, corresponding to the different Hyperswitch components.
```

> -  [App Server][app-server-repo] - The core payments engine responsible for managing payment flows, payment unification and smart routing, App server is maintained using this repo itself.
> 
> - [Web Client (SDK)][web-client-repo] - An inclusive, consistent and blended payment experience optimized for the best payment conversions
> 
> - [Control center][control-centre-repo] - A dashboard for payment analytics and operations, managing payment processors or payment methods and configuring payment routing rules.
> 
> [app-server-repo]: https://github.com/juspay/hyperswitch
> [web-client-repo]: https://github.com/juspay/hyperswitch-web
> [control-centre-repo]: https://github.com/juspay/hyperswitch-control-center

Reviewer: Nits:

```suggestion
-  [App Server][app-server] - The core payments engine responsible for managing payment flows, payment unification and smart routing. App server is maintained in this repo itself.

- [Web Client (SDK)][web-client] - An inclusive, consistent and blended payment experience optimized for the best payment conversions.

- [Control center][control-center] - A dashboard for payment analytics and operations, managing payment processors or payment methods and configuring payment routing rules.

[app-server]: https://github.com/juspay/hyperswitch
[web-client]: https://github.com/juspay/hyperswitch-web
[control-center]: https://github.com/juspay/hyperswitch-control-center
```

> ## Files Tree Layout

Reviewer: The top level directory structure looks different today, with some new folders and crates being present. I'm not sure how we should consider keeping this part up-to-date.

Author: This will be taken care in next iteration
